lib/chef/provider/aws_cloudwatch_alarm.rb

```diff
@@ -0,0 +1,48 @@
+require 'chef/provisioning/aws_driver/aws_provider'
+
+class Chef::Provider::AwsCloudwatchAlarm < Chef::Provisioning::AWSDriver::AWSProvider
+  provides :aws_cloudwatch_alarm
+
+  REQUIRED_OPTS = %i(namespace metric_name comparison_operator
+                     evaluation_periods period statistic threshold)
+
+  OTHER_OPTS = %i(dimensions insufficient_data_actions ok_actions
+                  actions_enabled alarm_actions alarm_description unit)
+
+  def update_aws_object(_instance)
+    Chef::Log.warn('aws_cloudwatch_alarm does not support modifying an alarm')
+  end
+
+  def create_aws_object
+    converge_by "creating cloudwatch alarm #{new_resource.name} in #{region}" do
+      new_resource.driver.cloudwatch.alarms.create(new_resource.name,
+                                                   options_hash)
+    end
+  end
+
+  def destroy_aws_object(_instance)
+    converge_by "destroying cloudwatch alarm #{new_resource.name} in #{region}" do
+      new_resource.driver.cloudwatch.alarms.delete(new_resource.name)
```

Comment:
Are the create and delete operations blocking? IE, will they wait to return until the cloudwatch alarm is created? Otherwise we need some waiters here